metricflow/test/model/semantics/test_join_validator.py

```diff
@@ -269,6 +274,83 @@ def test_foreign_target_instance_set_join_validation(
     )
 
 
+def test_get_joinable_data_sources(multi_hop_join_semantic_model: SemanticModel) -> None:  # noqa: D
+    data_source_reference = DataSourceReference(data_source_name="account_month_txns")
+    join_validator = DataSourceJoinValidator(data_source_semantics=multi_hop_join_semantic_model.data_source_semantics)
+
+    # Single-hop
+    joinable_data_sources = join_validator.get_joinable_data_sources(left_data_source_reference=data_source_reference)
+    assert set(joinable_data_sources.keys()) == {"bridge_table"}
+    assert joinable_data_sources["bridge_table"] == DataSourceLink(
+        left_data_source_reference=DataSourceReference(data_source_name="account_month_txns"),
+        join_path=[
+            DataSourceIdentifierJoin(
+                right_data_source_reference=DataSourceReference(data_source_name="bridge_table"),
+                identifier_reference=IdentifierReference(element_name="account_id"),
+                join_type=DataSourceIdentifierJoinType(
+                    left_identifier_type=IdentifierType.PRIMARY, right_identifier_type=IdentifierType.PRIMARY
+                ),
+            )
+        ],
+    )
+
+    # 2-hop
```

Comment:
nit: can we split the 1 hop and 2 hop tests into two test cases? That way we can see which ones are failing in the test runner.